playwright_test/utils/kmLocators.js

```diff
@@ -0,0 +1,22 @@
+
+/*
+ * kmLocators.js
+ * playwright_test/utils
+ * Created by Archit on 21/12/2011. 
+ */
+
+export const locators = {
+    envBtn : '//label[@for="test"]',
+    appIdField : '//*[@id="root"]/div/div/div[2]/div/div/input',
+    scriptFiled : '//*[@id="root"]/div/div/div[3]/div/div/textarea',
+    launchWidgetBtn : '(//button[@type="button"])[3]',
+    logoutWidgetBtn : '//button[text()="Logout & Reload"]'
+}
+
+export const widgetLocators = {
+    kmIframe : 'iframe[name="Kommunicate widget iframe"]',
+    kmLaunchWidget : '//*[@id="launcher-svg-container"]',
+    kmTextBox : '#mck-text-box',
+    kmSendButton : '//button[@title="Send Message"]//span[1]',
+    kmMsgStatus : '//*[@id="mck-message-cell"]/div[2]/div[1]/div/div[1]/div[2]/div[2]/div[4]/span[2]'
```

Comment:
same here

Reply:
@archit2602 One more thing we should create a package with name locators. there you should add all locators. so this is easy to use. util folder used for only utility things.

Reply:
Do you want me to create a new folder and move this locator file from util to that new folder right ???

Reply:
also @vipinsonu1, we can do anything with kmMsgStatus XPath because this is the only thing I have to get that

Reply:
check the nearest parent or child.

Reply:
this is the nearest one only 😅

Reply:
https://www.guru99.com/using-contains-sbiling-ancestor-to-find-element-in-selenium.html check this, other wise this will not work a long time.

Reply:
fixed thnx for this. 😊